optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/rotation/SpotLane.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.openshift.employeerostering.gwtui.client.pages.rotation;
+
+import java.util.List;
+
+import org.optaplanner.openshift.employeerostering.gwtui.client.beta.java.model.Lane;
+import org.optaplanner.openshift.employeerostering.gwtui.client.beta.java.model.SubLane;
+import org.optaplanner.openshift.employeerostering.shared.spot.Spot;
+
+public class SpotLane extends Lane<Long> {
+
+    private final Spot spot;
+
+    // Changing the parameter name to 'spot' leads to an error, because it uses the field instead of the parameter.
```

Comment:
How? It should default to the parameter, requiring you to use "this" to access the field.

Reply:
You can't write `this.spot` when performing a super constructor call.

Reply:
But you are using `spotParam` (the parameter) not spot (the member) (meaning you don't need this)

Reply:
If someone someday tries to change the parameter name to `spot`, it will cause an error, so I figured I could warn future readers to not do that :)